code/game/machinery/roulette_machine.dm

```diff
@@ -176,13 +177,24 @@
 				name = msg
 				desc = "Owned by [new_card.registered_account.account_holder], draws directly from [user.p_their()] account."
 				my_card = new_card
+				RegisterSignal(my_card, COMSIG_PARENT_QDELETING, .proc/on_my_card_deleted)
 				to_chat(user, span_notice("You link the wheel to your account."))
 				power_change()
 				return
 	return ..()
 
+///deletes the my_card ref to prevent harddels
+/obj/machinery/roulette/proc/on_my_card_deleted(datum/source)
+	SIGNAL_HANDLER
+	my_card = null
+
 ///Proc called when player is going to try and play
 /obj/machinery/roulette/proc/play(mob/user, obj/item/card/id/player_id, bet_type, bet_amount, potential_payout)
+	if(!my_card?.registered_account) // Something happened to my_card during the 0.4 seconds delay of the timed callback.
```

Comment:
why not do if(QDELLING(my_card)) return instead of the callbacks if you're concerned

Reply:
We can't just early return or the roulette will be stuck with the "rolling" icon state.